### PR TITLE
Pin boost to 1.74

### PR DESCRIPTION
**Pull request by phyy-nx**
conda currently picks 1.78, which requires c++14.  This is failing for dxtbx/dials.

See #2291

### CODE REVIEW

**Comment by graeme-winter** [comment]
Testing myself, now

**Comment by graeme-winter** [comment]
Worked for me: am tempted to merge-with-prejudice

File: newsfragments/2292.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Pin boost to 1.74
```
Comment (graeme-winter) [inline newsfragments/2292.bugfix:1]: Add comment linking to https://github.com/boostorg/geometry/issues/590 for background?

**Review by graeme-winter — APPROVED** [review]
LGTM

Wonder if we should add link to upstream change but other than that worked for me

**Comment by phyy-nx** [comment]
Thanks for the legwork @graeme-winter

**Pull request merged**